lib/engine/step/special_token.rb

```diff
@@ -94,7 +95,13 @@ def teleport_complete
       end
 
       def available_hex(entity, hex)
-        return if !ability(entity).hexes.empty? && !ability(entity).hexes.include?(hex.id)
+        ability = ability(entity)
+
+        if ability.type == :token && ability.reachable
+          return @game.token_graph_for_entity(entity.owner).reachable_hexes(entity.owner)[hex]
+        end
+
+        return if !ability.hexes.empty? && !ability.hexes.include?(hex.id)
```

Comment:
Move this back up to line 99. We need to check that the hex is valid before doing any further checks.